### PR TITLE
add round trip cbor test

### PR DESCRIPTION
**Pull request by flicknow**
adds tests for cbor serialization. currently fails

### CODE REVIEW

File: test/hexpds_dagcbor_test.exs [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
   doctest Hexpds.DagCBOR
+
+  defp test_cases do
+    [
+      "test text",
+      ["test text"],
+      %{string: "test text"},
+      %{map: %{string: "test text"}},
+      %{list: ["text"]},
+    ]
```
Comment (ShreyanJain9) [inline test/hexpds_dagcbor_test.exs]: do we know which ones work and which ones don't?

Comment (flicknow) [reply test/hexpds_dagcbor_test.exs]: only the first case of the simple string works right now

**Review by ShreyanJain9 — APPROVED** [review]

File: test/hexpds_dagcbor_test.exs (hunk 1 above)
Comment (ShreyanJain9) [reply test/hexpds_dagcbor_test.exs]: Cool, that's interesting. The IndexMap makes that fail since it's just a string, I assume?

Comment (flicknow) [reply test/hexpds_dagcbor_test.exs]: yep. same failures if i remove them and try with index amp
```
diff --git a/native/hexpds_dagcbor_internal/Cargo.toml b/native/hexpds_dagcbor_internal/Cargo.toml
index fa64d99..3a78552 100644
--- a/native/hexpds_dagcbor_internal/Cargo.toml
+++ b/native/hexpds_dagcbor_internal/Cargo.toml
@@ -10,6 +10,7 @@ path = "src/lib.rs"
 crate-type = ["cdylib"]

 [dependencies]
+indexmap = { version = "1.9.3", features = ["serde-1"] }
 rustler = "0.31.0"
 serde_ipld_dagcbor = "0.4.2"
 serde_json = "1.0.113"
diff --git a/native/hexpds_dagcbor_internal/src/lib.rs b/native/hexpds_dagcbor_internal/src/lib.rs
index 6d5d5d2..af78350 100644
--- a/native/hexpds_dagcbor_internal/src/lib.rs
+++ b/native/hexpds_dagcbor_internal/src/lib.rs
@@ -4,6 +4,7 @@ use rustler::Term;
 use rustler::Binary;
 use serde_json::from_str;
 use serde_ipld_dagcbor::to_vec;
+use indexmap::IndexMap;

 mod atoms {
     rustler::atoms! {
@@ -14,12 +15,9 @@ mod atoms {

 #[rustler::nif]
 fn encode_dag_cbor(env: Env, json: String) -> Term {
-        let parsed_json: serde_json::Value = match from_str(&json) {
-            Ok(json) => json,
-            Err(e) => return (atoms::error(), format!("Failed to parse JSON: {}", e)).encode(env),
-        };
+    let map_genesis: IndexMap<String, serde_json::Value> = serde_json::from_str(&json).unwrap();

-    let cbor_data = match to_vec(&parsed_json) {
+    let cbor_data = match to_vec(&map_genesis) {
         Ok(data) => data,
         Err(e) => return (atoms::error(), format!("Failed to encode to DAG-CBOR: {}", e)).encode(env),
     };
diff --git a/test/hexpds_dagcbor_test.exs b/test/hexpds_dagcbor_test.exs
index 631340e..8494870 100644
--- a/test/hexpds_dagcbor_test.exs
+++ b/test/hexpds_dagcbor_test.exs
@@ -4,8 +4,6 @@ defmodule HexpdsDagcborTest do

   defp test_cases do
     [
-      "test text",
-      ["test text"],
       %{"string" => "test text"},
       %{"map" => %{"string" => "test text"}},
       %{"list" => ["text"]},
```

still results in 
```
  1) test cbor roundtrip (HexpdsDagcborTest)
     test/hexpds_dagcbor_test.exs:13
     ** (MatchError) no match of right hand side value: {:error, "Failed to parse DAG-CBOR: TypeMismatch { name: \"CBOR tag\", byte: 2 }"}
     code: for input <- test_cases() do
     stacktrace:
       test/hexpds_dagcbor_test.exs:16: anonymous fn/2 in HexpdsDagcborTest."test cbor roundtrip"/1
       (elixir 1.16.1) lib/enum.ex:2528: Enum."-reduce/3-lists^foldl/2-0-"/3
       test/hexpds_dagcbor_test.exs:14: (test)
```

**Pull request merged**